### DISCUSSION TITLE
Correlated queries do not work well with nested columns

### DISCUSSION THREAD

**Discussion by xuanyu-z** [trinodb/trino#17712] [General]
A simplified example queries:

```
SELECT *
FROM   table_a AS a
WHERE  a.col_b IN (SELECT 1
                   FROM   table_a b
                   WHERE  col_a.field_a = a.col_a.field_a); 
```

The above query supposed to go through `TransformCorrelatedInPredicateToJoin` rule and the ApplyNode with its subquery supposed to be replaced with a left join.

However, possibly this change https://github.com/trinodb/trino/pull/7882 converted the column to a `subscriptExpression` and there are 2 extra projections added between the Filter and TableScan 
```
Project (Expr_X → Expr_X, Expr_Y → col_a[1])
Project (Expr_X → col_a_z[1])
```
This caused the visitor to determine it correlated shallowly and returned without a rewrite and left ApplyNode in the query plan, and fail the query with `Given correlated subquery is not supported` error.

Want to gain some insights on what is actually the root cause here?
are the extra projections supposed to be pushed down or removed?
or its the rules should handle the nested field properly after the subscriptExpression is being used instead of dereference symbol.

**Comment by findepi** [comment]
converted this to issue: https://github.com/trinodb/trino/issues/17714
